src/Sentry/SentryOptions.cs

```diff
@@ -461,14 +461,42 @@ public double TracesSampleRate
         /// </remarks>
         public Func<TransactionSamplingContext, double?>? TracesSampler { get; set; }
 
+        private StackTraceMode? _stackTraceMode;
+
         /// <summary>
         /// ATTENTION: This option will change how issues are grouped in Sentry!
         /// </summary>
         /// <remarks>
         /// Sentry groups events by stack traces. If you change this mode and you have thousands of groups,
         /// you'll get thousands of new groups. So use this setting with care.
         /// </remarks>
-        public StackTraceMode StackTraceMode { get; set; }
+        public StackTraceMode StackTraceMode
+        {
+            get
+            {
+                if (_stackTraceMode  is { } stackTraceMode)
+                {
+                    return stackTraceMode;
+                }
+
+                try
+                {
+                    // from 3.0.0 uses Enhanced (Ben.Demystifier) by default which is a breaking change
+                    // unless you are using .NET Native which isn't compatible with Ben.Demystifier.
+                    _stackTraceMode = Runtime.Current.Name == ".NET Native"
+                        ? StackTraceMode.Original
+                        : StackTraceMode.Enhanced;
+                }
+                catch (Exception ex)
+                {
+                    _stackTraceMode = StackTraceMode.Enhanced;
+                    DiagnosticLogger?.LogError($"Failed to get runtime, setting {nameof(StackTraceMode)} to {0} ", ex, _stackTraceMode);
```

Comment:
Avoid the interpolation, the generic overloads are here to avoid allocations (unless you measured and confirmed the compiler does that already):

```suggestion
                    DiagnosticLogger?.LogError("Failed to get runtime, setting {0} to {1} ", ex, nameof(StackTraceMode), _stackTraceMode);
```